lib/ShopifyTransform.ts

```diff
@@ -70,49 +124,62 @@ export class ShopifyTransform {
+        if (this.configuration.shop_sales !== true) {
+            // shipping
+            const shippingLine = this.shippingLines(sale, this.configuration.ecom_id)[0]
+            const shipping = shippingLine.behavior.shipping
+            const shippingAddress = shipping.address
+            const shippingCustomerInfo = shipping.customer_info
+            const parsedName = parsefullname.parseFullName(shippingAddress.name)
+
+            const shopifyShipping: any = {}
+            shopifyShipping.first_name = parsedName.first || ""
+            shopifyShipping.last_name = parsedName.last || ""
+            shopifyShipping.address1 = shippingAddress.street
+            shopifyShipping.city = shippingAddress.city
+            shopifyShipping.zip = shippingAddress.postal_code
+            shopifyShipping.country_code = shippingAddress.country_code || this.configuration.default_country_code
+            shopifyShipping.phone = shippingCustomerInfo.phone
+
+            if (shipping.method_id) {
+                const shopifyShippingLine: any = {
+                    code: shipping.method_id,
+                    price: shippingLine.total,
+                    title: shipping.method_id // Required field. Don't have anything better to put here unfortunately...
+                }
+                const taxes = this.shopifyTaxLines(shippingLine.taxes)
+                if (taxes.length > 0) {
+                    shopifyShippingLine.tax_lines = taxes
+                }
+                order.shipping_lines = [shopifyShippingLine]
             }
-            order.shipping_lines = [shopifyShippingLine]
-        }
 
-        order.shipping_address = shopifyShipping
-        order.email = shippingCustomerInfo.email
+            order.shipping_address = shopifyShipping
+            order.email = shippingCustomerInfo.email
+        } else {
+            // Shop sales
+            order.fulfillment_status = "fulfilled"
+            order.fulfillments = [
+                {
+                    "location_id": Number(locationId)
+                }
+            ]
+        }
 
         // line items
         const shopifyLineItems: any[] = []
-        for (const lineItem of this.ecommerceLines(sale, this.configuration.ecom_id)) {
+
+        const lineItems = (this.configuration.shop_sales === true) ? this.nonEcommerceLines(sale) : this.ecommerceLines(sale, this.configuration.ecom_id)
+
+        for (const lineItem of lineItems) {
             let variantId: string | undefined = lineItem.variant_id
             if (_.isNil(variantId)) {
                 try {
-                    const shopifyProduct = await this.shopifyProduct(lineItem.id, this.configuration)
-                    if (shopifyProduct && 
-                        shopifyProduct.product && 
-                        shopifyProduct.product.variants && 
-                        shopifyProduct.product.variants[0] && 
+                    const shopifyProduct = await this.shopifyProduct(lineItem.id)
```

Comment:
I assume there isn't a way to do optional unwrapping that doesn't require this traversal through the structure?